{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1>KEGG decoder examples and usage</h1>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "    This Github repository is designed to generate a heatmap of functional pathways for a set of genomes and compare the clustering of these genomes to a taxonomic tree by generating a tanglegram."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "    It is based off of scripts and work done by Ben Tully, available at: https://github.com/bjtully/BioData/tree/master/KEGGDecoder and modified by Ben Tully, Roth Conrad, Taylor Reitner, Luiz Irber, and Jay Osvatic.\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<strong>Input file generation and formatting for functional heatmap </strong>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The functional heatmaps are generated by the default output of BlastKOALA, GhostKOALA or KOFAMscan?\n",
    "\n",
    "The ID of each genome must be before an _ in the name of the protien , i.e. \"GENOME1_1\" or \"GENOME1_2\". This will allow for the proper categorization of the functional protiens in genomes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<strong>Input format</strong>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!head Example_input_data.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<strong>Functional heatmap generation</strong>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "    For a static heatmap:\n",
    "    \n",
    "        !KEGG-decoder --input INPUT.txt --output OUTPUT.tsv --vizoption static\n",
    "        \n",
    "    Here is our example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!KEGG-decoder --input Example_input_data.txt --output functions.tsv --vizoption static"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "    For a interactive heatmap, with 4 clustering options (Euclidean, Correlation, In descending order of # genes annotated, In ascending order of # genes annotated):\n",
    "    \n",
    "        !KEGG-decoder --input INPUT.txt --output OUTPUT.tsv --vizoption interactive\n",
    "        \n",
    "        This will generate an html called \"@@@@@@@@@@\"\n",
    "        \n",
    "    Here is our example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!KEGG-decoder --input Example_input_data.txt --output functions.tsv --vizoption interactive"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<strong>Input file generation and formatting for tanglegram </strong>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
